### ISSUE TITLE
Issue adding BID 

### ISSUE THREAD

**Issue by dhoer** [univ-of-utah-marriott-library-apple/privacy_services_manager#30]
Not able to add `BID`:
`com.apple.RemoteDesktopAgent`

But I am able to add `Path`:
`/System/Library/CoreServices/RemoteManagement/ARDAgent.app`

You can infer what I'm passing in from this:
```
privacy_services_manager 'make rdagent accessible' do
  service 'accessibility'
  user 'vagrant'
  applications ['/System/Library/CoreServices/RemoteManagement/ARDAgent.app', 
                '/usr/libexec/sshd-keygen-wrapper']
  admin true
end
```

```
$ sudo /usr/local/bin/privacy_services_manager.py --user vagrant --admin add accessibility /System/Library/CoreServices/RemoteManagement/ARDAgent.app
$ sudo /usr/local/bin/privacy_services_manager.py --user vagrant --admin add accessibility /usr/libexec/sshd-keygen-wrapper
```

```
$ app_lookup.py com.apple.RemoteDesktopAgent
ARDAgent
	BID:        com.apple.RemoteDesktopAgent
	Path:       /System/Library/CoreServices/RemoteManagement/ARDAgent.app
	Info.plist: /System/Library/CoreServices/RemoteManagement/ARDAgent.app/Contents/Info.plist
	Executable: /System/Library/CoreServices/RemoteManagement/ARDAgent.app/Contents/MacOS/ARDAgent
```

**Comment by pdarragh** [comment]
Hmm... Are you also using the `admin` flag when you're trying to add by `BID`? If so, then technically this is an expected outcome, though maybe not an intuitive one.

The `admin` flag specifically tells the script to assume that the given value is a path to an executable, meaning that it doesn't even try to check whether it's a bundle identifier or a regular name first. When I added the `admin` flag, I figured it would only be used on a case-by-case basis.

I think I want to change that, though. I'll try to go through and update the script so it will attempt to match `BID` or name first, and then – only then – if it fails, it'll utilize the administrative override if it was enabled. Does that seem like it would make more sense, you think?
